### ISSUE TITLE
(yarn install in exmaple) error An unexpected error occurred: "Reduce of empty array with no initial value".

### ISSUE THREAD

**Issue by shinjikao** [tabler/tabler-react#191]
When I want following instruction  install and setup to run this tabler-react, I got trouble when yarn install in example folder. it show up error on screen.

error An unexpected error occurred: "Reduce of empty array with no initial value"

Arguments: 
  /usr/local/bin/node /usr/local/Cellar/yarn/1.7.0/libexec/bin/yarn.js install

PATH: 
  /Users/jackalkao/anaconda3/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin

Yarn version: 
  1.7.0

Node version: 
  8.11.2

Platform: 
  darwin x64

Trace: 
  TypeError: Reduce of empty array with no initial value
      at Array.reduce (<anonymous>)
      at PackageLinker.<anonymous> (/usr/local/Cellar/yarn/1.7.0/libexec/lib/cli.js:50461:43)
      at Generator.next (<anonymous>)
      at step (/usr/local/Cellar/yarn/1.7.0/libexec/lib/cli.js:98:30)
      at /usr/local/Cellar/yarn/1.7.0/libexec/lib/cli.js:109:13
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:188:7)

npm manifest: 
  {
    "name": "tabler-react-example",
    "homepage": "https://jonthomp.github.io/tabler-react",
    "version": "0.0.0",
    "private": true,
    "license": "MIT",
    "dependencies": {
      "d3-scale": "^2.0.0",
      "prop-types": "^15.6.1",
      "react": "^16.2.0",
      "react-c3js": "^0.1.20",
      "react-dom": "^16.2.0",
      "react-google-maps": "^9.4.5",
      "react-router-dom": "^4.2.2",
      "react-scripts": "^1.1.1",
      "react-simple-maps": "^0.12.0",
      "react-syntax-highlighter": "^7.0.2",
      "tabler-react": "link:.."
    },
    "scripts": {
      "start": "react-scripts start",
      "watch": "react-scripts start --watch",
      "build": "react-scripts build",
      "test": "react-scripts test --env=jsdom",
      "eject": "react-scripts eject",
      "predeploy": "npm run build",
      "deploy": "gh-pages -d build"
    },
    "devDependencies": {
      "bootstrap": "^4.1.0",
      "jquery": "^3.3.1",
      "popper.js": "^1.14.3"
    }
  }

yarn manifest: 
  No manifest

**Comment by jonthomp** [comment]
Hi @shinjikao this looks like an [issue with yarn](https://github.com/yarnpkg/yarn/issues/5876). There is fix available on the [nightly version](https://yarnpkg.com/lang/en/docs/nightly/) or you could try this suggestion from the issue comments:

In example folder run:
```
rm -rf node_modules
yarn
yarn link tabler-react
```

**Comment by shinjikao** [comment]
@jonthomp  no working

**Comment by shinjikao** [comment]
switch to yarn 1.6 can fix my problem. work fine.

$ brew unlink yarn

$ brew install https://raw.githubusercontent.com/Homebrew/homebrew-core/76215230de5f7f7bee2cfcdd7185cf49d949862d/Formula/yarn.rb

$ brew switch yarn 1.6.0_1

**Comment by AaronCoplan** [comment]
Glad it's working!

**Comment by samccone** [comment]
A note, this bug reproduced with yarn `1.7.0`

yarn link SOME_PACKAGE
yarn add --dev SOME_OTHER_PACKAGE

**Comment by dvisztempacct** [comment]
I can also report I have been experiencing this problem on `1.7.0` as well. I keep doing `rm -fr node_modules && yarn` to address the issue. I might be getting this error after using `yarn link` I'm not sure. It's getting quite annoying!

**Comment by jonthomp** [comment]
@samccone @dvisztempacct sorry I'm afraid this appears to be out of our hands as it's a yarn issue that many have had in other projects https://github.com/yarnpkg/yarn/issues/5876, are you able to update from 1.7.0?

**Comment by dvisztempacct** [comment]
Thanks @jonthomp :)

**Comment by bilo-io** [comment]
I was experiencing the same error message when running `yarn`. My scenario, and how I fixed it, if this helps anyone:

1. I had a locally linked dependency in my project (`yarn link {package_name}`)
2. I was pointing to a github branch of that dependency in `package.json`
3. when running `yarn` I got the error message: `An unexpected error occurred: "Reduce of empty array with no initial value".`

SOLUTION:

1. Unlink the package `yarn unlink {package_name}`
2. run `yarn`
